nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage.java

```diff
@@ -91,6 +124,15 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                 }
 
                 try {
+                    if (!userMetadata.isEmpty()) {
+                        blob.setMetadata(userMetadata);
+
+                        StringBuilder userMetaBldr = new StringBuilder();
+                        for (String userKey : userMetadata.keySet()) {
+                            userMetaBldr.append(userKey).append("=").append(userMetadata.get(userKey));
+                        }
```

Comment:
shouldn't we have a delimiter between entries? For example, if we have these properties:
runner.setProperty("hello", "world");
runner.setProperty("java", "world");

then, azure.usermetadata='java=worldhello=world' when I modify and run your test.

Reply:
This code is 1:1 stolen from the s3 processor.If we make changes here maybe we should update this one in following pull request to.
This is in the end only debug stuff, more or less.
But will look into this

Reply:
This should be fixed now. (only for the new code, if this is fine I will create a new Pull Request for s3)